### PR TITLE
Upstream merge

### PR DESCRIPTION
**Pull request by TsjipTsjip**
Merges upstream up to and including 1faa1b5df6e74b9cb1309d0400b12ee0b96039e4 .

Attention points: Buckling.

Also merges CD including:
- https://github.com/cosmatic-drift-14/cosmatic-drift/pull/314
- https://github.com/cosmatic-drift-14/cosmatic-drift/pull/316
- https://github.com/cosmatic-drift-14/cosmatic-drift/pull/318

### CODE REVIEW

**Comment by arimah** [comment]
Question: What is the point of the QuickThreads restock crate? We never play on any stations with a QuickThreads – it's only on the arrivals map, which no one ever goes back to.

**Comment by TsjipTsjip** [comment]
> Question: What is the point of the QuickThreads restock crate? We never play on any stations with a QuickThreads – it's only on the arrivals map, which no one ever goes back to.

Allows us to restock the QuickThreads at the terminal, which people do actually go back to sometimes, both IC and OOC depending on the needs of the station.

File: Content.Server/_CD/Chemistry/ReagentEffects/ResetParacusia.cs [hunk 1]
```diff
@@ -1,26 +1,29 @@
...
     public int TimerReset = 600;
 
     protected override string? ReagentEffectGuidebookText(IPrototypeManager prototype, IEntitySystemManager entSys)
         => Loc.GetString("reagent-effect-guidebook-reset-paracusia", ("chance", Probability));
 
-    public override void Effect(ReagentEffectArgs args)
+    public override void Effect(EntityEffectBaseArgs args)
     {
-        if (args.Scale != 1f)
+        EntityEffectReagentArgs rArgs = (EntityEffectReagentArgs) args;
```
Comment (arimah) [inline Content.Server/_CD/Chemistry/ReagentEffects/ResetParacusia.cs:21]: Don't use a cast. If the cast fails, you will cause an exception, which, above all else, can really tank performance. For consistency with other code, consider something like this:

```csharp
if (args is not EntityEffectReagentArgs reagentArgs)
    return;
```

**Review by Simyon264 — APPROVED** [review]

**Review by arimah — APPROVED** [review]
Everything seems to work fine in testing. Can't find any glaring issues.

**Pull request merged**